Lib/base64.py

```diff
@@ -490,25 +490,16 @@ def b85decode(b):
 
 def encode(input, output):
     """Encode a file; input and output are binary files."""
-    while True:
-        s = input.read(MAXBINSIZE)
-        if not s:
-            break
-        while len(s) < MAXBINSIZE:
-            ns = input.read(MAXBINSIZE-len(s))
-            if not ns:
-                break
+    while (s := input.read(MAXBINSIZE)):
+        while len(s) < MAXBINSIZE and (ns := input.read(MAXBINSIZE-len(s))):
             s += ns
         line = binascii.b2a_base64(s)
         output.write(line)
 
 
 def decode(input, output):
     """Decode a file; input and output are binary files."""
-    while True:
-        line = input.readline()
-        if not line:
-            break
+    while (line := input.readline()):
```

Comment:
i would like to make a point of just using an actual for loop for those case simply not to litter with missuses of the feature where applicable

Reply:
"for line in input:" and "while (line := readline()):" are not strictly the same: input iterator can use something different than readline...